lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -85,50 +90,91 @@ export default function LMSFilePicker({
   onCancel,
   onSelectFile,
   missingFilesHelpLink,
+  withBreadcrumbs = false,
 }) {
   // The main state of the dialog and associated data.
   const [dialogState, setDialogState] = useState(INITIAL_DIALOG_STATE);
 
   // Authorization attempt was made. Set after state transitions to "authorizing".
   const [authorizationAttempted, setAuthorizationAttempted] = useState(false);
 
-  // The file within `files` which is currently selected.
-  const [selectedFile, selectFile] = useState(/** @type {File|null} */ (null));
+  // The file or folder within `files` which is currently selected.
+  const [selectedFile, setSelectedFile] = useState(
+    /** @type {File|null} */ (null)
+  );
+
+  // An array of File objects representing the  "path" to the current
+  // list of files being displayed. This always starts at the root. The last
+  // element represents the current directory path.
+  const [folderPath, setFolderPath] = useState(
+    /** @type File[] */ ([
```

Comment:
Ah - it looks like the `{ ... }` around the type name is not strictly required here. If that is behavior we can rely on it might make sense to prefer this syntactically less-noisy form of annotation in future.

Reply:
On the down side, my editor doesn't do syntax highlighting without the `{..}`, so I'm adding it in for now.